Documentation/vmware-backdoor.md

```diff
@@ -0,0 +1,26 @@
+# VMware Backdoor #
+coreos-cloudinit is capable of reading userdata and metadata from the VMware
+backdoor. This datasource can be enable with the `--from-vmware-backdoor` flag.
+Userdata and metadata are passed from the hypervisor to the virtual machine
+through guest variables. The following guest variables and their expected types
+are supported by coreos-cloudinit:
+
+|          guest variable           |              type               |
+|:----------------------------------|:--------------------------------|
+| `hostname`                        | `hostname`                      |
+| `interfaces.<n>.ipv4.<m>.proto`   | `{"dhcp", "static"}`            |
+| `interfaces.<n>.ipv4.<m>.address` | `IPv4 address`                  |
+| `interfaces.<n>.ipv4.<m>.netmask` | `IPv4 netmask`                  |
+| `interfaces.<n>.ipv6.<m>.proto`   | `{"dhcp", "static"}`            |
+| `interfaces.<n>.ipv6.<m>.address` | `IPv6 address`                  |
+| `interfaces.<n>.ipv6.<m>.cidr`    | `int`                           |
+| `interfaces.<n>.role`             | `{"public", "private"}`         |
+| `interfaces.<n>.mac`              | `MAC address`                   |
+| `interfaces.<n>.gateway`          | `IP address`                    |
```

Comment:
Well that's just depressing that the whoever came up with this list didn't consider both v4 and v6 or other routes but ok. I suppose it is assumed that v6 always uses router discovery even if the address itself is static.